[3.0] Public Cloud: Add support for bootstrapping in Google Compute

bear454 opened the pull request:
(cherry picked from commit 446f412a3fd871bfae8429d9131fe462cbbb0d5e)

bear454 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Public Cloud: Add support for bootstrapping in Google Compute](https://trello.com/c/N0jyJj5v/1051-public-cloud-add-support-for-bootstrapping-in-google-compute)

MaximilianMeister commented:
@bear454 can you make sure to get the tests green here?

bear454 commented:
Sorry @MaximilianMeister - I thought `product_name` was defined for 3.x.

MaximilianMeister commented:
LGTM :+1: 

@jordimassaguerpla is it ok like this or should we squash the commits or add the bug reference to the 2nd commit as well? in jenkins we test that every backport commit has a bug reference.

@bear454 feel free to squash the commits, or add the bug reference to the 2nd commit

MaximilianMeister commented:
> I thought product_name was defined for 3.x.

yeah as we havent backported the branding this was not really obvious

bear454 commented:
I've squashed the commit, but there's no bugzilla entry for adding public cloud support, @jordimassaguerpla .

@MaximilianMeister , please reset your review; github removed it when I squashed the commit.

jordimassaguerpla commented:
@bear454 : fine with me. We will figure out during the packaging process if we have to add some reference.

bear454 commented:
Rebased, and added in the necessary fixes for BSC#1098085, as @kiall requested.

bear454 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Bug 1098085 - No UI for accessing node bootstrap info after setup](https://trello.com/c/JBsPkG6f/1086-bug-1098085-no-ui-for-accessing-node-bootstrap-info-after-setup)

kiall commented:
@MaximilianMeister - can you double check? Should be good to go though.

MaximilianMeister commented on app/controllers/setup_controller.rb:
```diff
@@ -11,7 +11,8 @@ class SetupController < ApplicationController
   before_action :redirect_to_dashboard
-  skip_before_action :redirect_to_dashboard, only: :worker_bootstrap
+  skip_before_action :redirect_to_dashboard,
+                     only: [:worker_bootstrap, :build_cloud_cluster]
```
:+1:
